src/viking.js

```diff
@@ -70,35 +72,41 @@ War.prototype.addSaxon = function(saxon) {
 };
 
 War.prototype.vikingAttack = function() {
+  // determine random index of both viking and saxon.
   const targetVikingIndex = [Math.floor(Math.random() * this.vikingArmy.length)];
   const targetSaxonIndex = [Math.floor(Math.random() * this.saxonArmy.length)];
-  this.saxonArmy[targetSaxonIndex].receiveDamage(
-    this.vikingArmy[targetVikingIndex].attack()
-  );
-  if (this.saxonArmy[targetSaxonIndex].health < 0) {
+  // viking attack on the saxon
+  this.saxonArmy[targetSaxonIndex].receiveDamage(this.vikingArmy[targetVikingIndex].attack());
+  // remove saxon from army if it dies + message
+  if (this.saxonArmy[targetSaxonIndex].health <= 0) {
     this.saxonArmy.splice(targetSaxonIndex, 1);
     return "A Saxon has died in combat";
   }
+  // output message for the attack
+  return `A Saxon has received ${this.saxonArmy[targetSaxonIndex].strength} points of damage`;
 };
 
 War.prototype.saxonAttack = function() {
+  // determine random index of both viking and saxon.
   const targetVikingIndex = [Math.floor(Math.random() * this.vikingArmy.length)];
   const targetSaxonIndex = [Math.floor(Math.random() * this.saxonArmy.length)];
-  this.vikingArmy[targetVikingIndex].receiveDamage(
-    this.saxonArmy[targetSaxonIndex].attack()
-  );
-  if (this.vikingArmy[targetVikingIndex].health < 0) {
+  // saxon attack on the viking
+  this.vikingArmy[targetVikingIndex].receiveDamage(this.saxonArmy[targetSaxonIndex].attack());
+  // remove viking from army if it dies + message
+  if (this.vikingArmy[targetVikingIndex].health <= 0) {
     this.vikingArmy.splice(targetVikingIndex, 1);
+    return `${this.vikingArmy[targetVikingIndex].name} has died in combat`;
```

Comment:
You can't call the name of the viking you just delete because it doesn't exist anymore... so you need to save the data in a variable to be able to return it